### PR TITLE
Module generation cli 

### PR DESCRIPTION
**Pull request by gloriacai01**
Add `module generate` option to viam cli
`module generate` will show a form which asks users for input on information about their module, and then generates all the needed files + method stubs
Right now, only supports python module generation

### CODE REVIEW

File: cli/templates/.viam-gen-info
Comment (njooma) [inline cli/templates/.viam-gen-info:1]: Delete this file

Comment (njooma) [reply cli/templates/.viam-gen-info:1]: Delete this whole directory even

File: .github/workflows/pullrequest-trusted.yml
Comment (njooma) [inline .github/workflows/pullrequest-trusted.yml:1]: We probably should revert this file

File: .github/workflows/test.yml
Comment (njooma) [inline .github/workflows/test.yml:1]: @abe-winter pls confirm that this is ok?

File: cli/module_generate/templates/python/tmpl-requirements.txt [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{{ if eq .ResourceSubtype "mlmodel" }}
+viam-sdk[mlmodel]
```
Comment (njooma) [inline cli/module_generate/templates/python/tmpl-requirements.txt:2]: @purplenicole730 we still want to pin this to `{{ .SDKVersion }}`

**Review by njooma — CHANGES_REQUESTED** [review]

File: cli/module_generate/templates/python/tmpl-requirements.txt (hunk 1 above)
Comment (purplenicole730) [reply cli/module_generate/templates/python/tmpl-requirements.txt:2]: Thanks for catching that! I'll fix that in my next PR

**Comment by njooma** [comment]
@abe-winter I added you as a reviewer specifically to take a look at the test.yaml workflow to make sure we're doing the python stuff correctly. feel free to ignore the rest of it (or review the rest of it, i don't control you)

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -39,6 +39,10 @@ jobs:
       with:
         ref: ${{ github.event_name == 'pull_request_target' && github.event.pull_request.head.sha || github.event.ref }}
 
+    - uses: actions/setup-python@v5
```
Comment (abe-winter) [inline .github/workflows/test.yml:42]: this is a no-op (I know that's not inuitive) + is easy to break in the future; would remove

**Review by abe-winter — APPROVED** [review]
test.yml changes LGTM but see 1 comment

File: .github/workflows/test.yml (hunk 2 above)
Comment (njooma) [reply .github/workflows/test.yml]: What about about for catching pip packages?

Comment (abe-winter) [reply .github/workflows/test.yml]: the `cache: pip` stanza keys on a `requirements.txt` file and crashes if it can't find one

the reason it's not crashing in our case is that we're not passing `with: python-version`, so the action is defaulting to the system python + short-circuiting

Comment (njooma) [reply .github/workflows/test.yml]: Cool, removed setup python and apt updates

**Review by purplenicole730 — APPROVED** [review]
LGTM!

File: cli/module_generate.go [hunk 3]
```diff
@@ -0,0 +1,726 @@
...
+		return errors.Wrapf(err, "failed to create %s", infoFilePath)
+	}
+	defer utils.UncheckedErrorFunc(infoFile.Close)
+
+	if _, err := infoFile.Write(infoBytes); err != nil {
+		return errors.Wrapf(err, "failed to write generator info to %s", infoFilePath)
+	}
+
+	// Render workflows for cloud build
+	if module.EnableCloudBuild {
```
Comment (stuqdog) [inline cli/module_generate.go:397]: (opt) this block of code gets pretty nested and that makes it pretty hard to follow. I think some refactoring could definitely be done to clean up the control flow here, but it might be a bit of a hassle to do. At a minimum though, I think we could move everything one level up by inverting this, i.e.,

```go
if !module.EnableCloudBuild {
    return
}
...
```

File: cli/module_generate.go [hunk 4]
```diff
@@ -0,0 +1,726 @@
...
+			if err != nil {
+				return err
+			}
+		}
+		return nil
+	})
+	if err != nil {
+		return err
+	}
+	return nil
```
Comment (stuqdog) [inline cli/module_generate.go:555]: (minor) I think this can be simplified to `return err`

File: cli/module_generate.go [hunk 5]
```diff
@@ -0,0 +1,726 @@
...
+	}
+
+	return nil
+}
+
+func getLatestSDKTag(c *cli.Context, language string) (string, error) {
+	var repo string
+	if language == python {
+		repo = "viam-python-sdk"
+	}
```
Comment (stuqdog) [inline cli/module_generate.go:614]: (minor, opt) probably good for this to be a switch statement for future extensibility.

File: cli/module_generate/templates/python/.gitignore
Comment (stuqdog) [inline cli/module_generate/templates/python/.gitignore:1]: (q) this `.gitignore` seems super comprehensive! Did we crib this from somewhere? If so, should we be providing attribution at the top?

File: cli/module_generate.go (hunk 3 above)
Comment (njooma) [reply cli/module_generate.go:397]: I fell into a trap I hate. I purposefully didn't return early because this renders all common files, so I wanted to be able to leave space for additional common files that come after this. If we return early, that doesn't allow for adding new files afterward without changing the code again. 

But that's what you call "premature optimization"

File: cli/module_generate/templates/python/.gitignore
Comment (njooma) [reply cli/module_generate/templates/python/.gitignore:1]: It's from the git ignores github `https://github.com/github/gitignore/blob/main/Python.gitignore`

File: cli/module_generate.go [hunk 6]
```diff
@@ -0,0 +1,726 @@
...
+					if _, err := os.Stat(s); err == nil {
+						return errors.New("this module directory already exists")
+					}
+					return nil
+				}),
+			huh.NewSelect[string]().
+				Title("Specify the language for the module:").
+				Options(
+					huh.NewOption("Python", python),
+					// huh.NewOption("Go", "go"),
```
[the remark this answers is not in the document]
Comment (njooma) [reply cli/module_generate.go:232]: I agree, but want to keep this in and not change it because i know there's work going on in golang now that would uncomment this line, and that would make for an annoying merge commit since both of these lines would be changed :( but good to note and I agree for future

**Review by stuqdog — APPROVED** [review]
